app/src/main/java/de/westnordost/streetcomplete/quests/shop_type/CheckShopType.kt

```diff
@@ -48,31 +48,26 @@ class CheckShopType : OsmElementQuestType<ShopTypeAnswer> {
 
     override fun createForm() = ShopTypeForm()
 
-    override fun applyAnswerTo(answer: ShopTypeAnswer, changes: StringMapChangesBuilder) {
+    override fun applyAnswerTo(answer: ShopTypeAnswer, tags: StringMapChangesBuilder) {
 
         when (answer) {
             is IsShopVacant -> {
-                changes.updateCheckDate()
+                tags.updateCheckDate()
             }
             is ShopType -> {
-                changes.deleteCheckDates()
-
-                if (!answer.tags.containsKey("shop")) {
-                    changes.deleteIfExists("shop")
-                }
-
-                for ((key, _) in changes.getPreviousEntries()) {
+                for (key in tags.keys) {
                     // also deletes all "disused:" keys
-                    val isOkToRemove =
-                        KEYS_THAT_SHOULD_BE_REMOVED_WHEN_SHOP_IS_REPLACED.any { it.matches(key) }
+                    val isOkToRemove = KEYS_THAT_SHOULD_BE_REMOVED_WHEN_SHOP_IS_REPLACED.any { it.matches(key) }
                     if (isOkToRemove && !answer.tags.containsKey(key)) {
```

Comment:
No, this will spawn an IllegalStateException in the StringMapChangesBuilder. Currently, the StringMapChangesBuilder checks that each key is only modified once. E.g.
```
tags["a"] = "b"
tags["a"] = "c"
```
currently throws. Sure, I could remove this check and instead overwrite the change made, but I was thinking that with this check, maybe unintended behavior in the code could be spotted.